.github/scripts/modules-to-build.py

```diff
@@ -1,74 +1,68 @@
+import itertools
+import json
 import os
-import subprocess as sp
 import re
-import json
+import subprocess
 import sys
 from pathlib import Path
-from typing import List, Tuple, Dict, Any
+from typing import NoReturn
+
+EXTENSION_REGEX = re.compile(r"^src/(?P<lang>\w+)/(?P<extension>\w+)")
+MULTISRC_LIB_REGEX = re.compile(r"^lib-multisrc/(?P<multisrc>\w+)")
+LIB_REGEX = re.compile(r"^lib/(?P<lib>\w+)")
+
+def run_command(command: str) -> str:
+    result = subprocess.run(command, capture_output=True, text=True)
+    if result.returncode != 0:
+        print(result.stderr.strip())
+        sys.exit(result.returncode)
+    return result.stdout.strip()
+
+def get_module_list(ref: str) -> tuple[list[str], list[str]]:
+    changed_files = run_command(f"git diff --name-only {ref}").splitlines()
 
-def get_module_list(commitHash: str) -> Tuple[List[str], List[str]]:
-    file_list = sp.run(
-        ["git", "diff", "--name-only", commitHash],
-        check=True,
-        capture_output=True,
-        text=True
-    ).stdout
-            
     modules = set()
     libs = set()
     deleted = set()
-    
-    ext_regex = re.compile(r"^src/(?P<lang>\w+)/(?P<ext_name>\w+)")
-    multisrc_regex = re.compile(r"^lib-multisrc/(?P<multisrc>\w+)")
-    libs_regex = re.compile(r"^lib/(?P<lib>\w+)")
 
-    for _file in file_list.splitlines():
-        file = Path(_file).as_posix()
-                
-        ext_match = ext_regex.search(file)
-        multisrc_match = multisrc_regex.search(file)
-        lib_match = libs_regex.search(file)
-        
-        if ext_match:
-            lang = ext_match.group("lang")
-            ext_name = ext_match.group("ext_name")
-            if Path("src", lang, ext_name).is_dir():
-                modules.add(f':src:{lang}:{ext_name}')
-            deleted.add(f"{lang}.{ext_name}") # need to add here so we can delete old version during commit stage
-        elif multisrc_match:
-            multisrc = multisrc_match.group("multisrc")
+    for file in map(lambda x: Path(x).as_posix(), changed_files):
+        if match := EXTENSION_REGEX.search(file):
+            lang = match.group("lang")
+            extension = match.group("extension")
+            if Path("src", lang, extension).is_dir():
+                modules.add(f':src:{lang}:{extension}')
+                deleted.add(f"{lang}.{extension}")
```

Comment:
```suggestion
            deleted.add(f"{lang}.{extension}")
```
when an extension is removed from the repo, we need to delete it's corresponding apk too

Reply:
hmm right

Reply:
don;t we also need to remove it in case of multisrc and potentially lib?

Reply:
the assumption is that if a multisrc or lib is removed, it's corresponding extension will have ben removed alongside, otherwise it will just be build error

Reply:
but in case they are updated won't we need to remove the old associated extensions

Reply:
yes we would. ig the modules to build and deleted both should have same set of extensions...